packages/core/src/Operation.ts

```diff
@@ -22,32 +23,22 @@ export interface Operation<I extends Product, O extends Product> {
 }
 
 export interface OperationOptions {
-  readonly hierarchy?: Hierarchy,
+  readonly hierarchy?: Promisish<Hierarchy | undefined>,
   readonly overrideHierarchy?: boolean,
   readonly showOutputInHierarchy?: boolean,
 }
 
 export const Operation = {
   create: <I extends Product, O extends Product>(
     name: string,
-    func: (arg: Element<I>) => Elementish<O>,
+    func: (arg: Element<I>) => Promisish<Elementish<O>>,
```

Comment:
consider extracting type to `Element.ts`

Reply:
`Elementish` can't have `Promisish` baked in, as it was causing the inferred type of `Element` to be too large. I feel that `PromisishElementish` adds little, and other names would be confusing (e.g. `AsyncElementish` suggests that `Elementish` is fully synchronous), so I'm going to leave this in. It's only used in 3 places.